File: apps/chat/system-server/src/main/java/com/example/chat/system/application/dto/response/ChannelMetadataResponse.java
package com.example.chat.system.application.dto.response;

import java.time.Instant;

import com.example.chat.domain.channel.metadata.ChannelMetadata;

/**
 * 채팅방 메타데이터 Response DTO
 */
public record ChannelMetadataResponse(
        String id,
        String channelId,
        String userId,
        boolean notificationEnabled,
        boolean favorite,
        boolean pinned,
        String lastReadMessageId,
        Instant lastReadAt,
        int unreadCount,
        Instant lastActivityAt,
        Instant createdAt,
        Instant updatedAt) {
    public static ChannelMetadataResponse from(ChannelMetadata metadata) {
        return new ChannelMetadataResponse(
                metadata.getId().value(),
                metadata.getChannelId().value(),
                metadata.getUserId().value(),
                metadata.isNotificationEnabled(),
                metadata.isFavorite(),
                metadata.isPinned(),
                metadata.getLastReadMessageId() != null ? metadata.getLastReadMessageId().value() : null,
                metadata.getLastReadAt(),
                metadata.getUnreadCount(),
                metadata.getLastActivityAt(),
                metadata.getCreatedAt(),
                metadata.getUpdatedAt());
    }
}

File: apps/chat/system-server/src/main/kotlin/com/example/chat/system/application/dto/response/ChannelMetadataResponse.kt
package com.example.chat.system.application.dto.response

import com.example.chat.domain.channel.metadata.ChannelMetadata
import java.time.Instant

/**
 * 채팅방 메타데이터 Response DTO
 */
data class ChannelMetadataResponse(
	val id: String,
	val channelId: String,
	val userId: String,
	val notificationEnabled: Boolean,
	val favorite: Boolean,
	val pinned: Boolean,
	val lastReadMessageId: String?,
	val lastReadAt: Instant?,
	val unreadCount: Int,
	val lastActivityAt: Instant,
	val createdAt: Instant,
	val updatedAt: Instant
) {
	companion object {
		fun from(metadata: ChannelMetadata) = ChannelMetadataResponse(
			id = metadata.id.value,
			channelId = metadata.channelId.value,
			userId = metadata.userId.value,
			notificationEnabled = metadata.notificationEnabled,
			favorite = metadata.favorite,
			pinned = metadata.pinned,
			lastReadMessageId = metadata.lastReadMessageId?.value,
			lastReadAt = metadata.lastReadAt,
			unreadCount = metadata.unreadCount,
			lastActivityAt = metadata.lastActivityAt,
			createdAt = metadata.createdAt,
			updatedAt = metadata.updatedAt
		)
	}
}

/**
 * Extension Function
 */
fun ChannelMetadata.toResponse() = ChannelMetadataResponse.from(this)
